kraken_frontend/src/utils/filter/parser.ts

```diff
@@ -382,50 +88,56 @@ function parseAst(input: string, parseColumn: (column: string, cursor: Cursor) =
 }
 
 /** Parse an {@link Expr.Or} expression using a `parseValue` to parse the leaves */
-function parseOr<T>(tokens: Cursor, parseValue: (cursor: Cursor) => Expr.Value<T>): Expr.Or<T> {
-    const list = [parseAnd(tokens, parseValue)];
-    while (tokens.peekToken()?.type === "logicalOr") {
-        tokens.nextToken(); // Consume the `,`
-        list.push(parseAnd(tokens, parseValue));
-    }
-    return { or: list };
+export function parseOr<T>(parseValue: (cursor: Cursor) => Expr.Value<T>): (tokens: Cursor) => Expr.Or<T> {
```

Comment:
why does this return a lambda?
It is used to wrap every `parseValue` in the declaration of the asts.
But there it is unconditionally wrapped around every single field.
So why don't remove the `parseOr` entirely from the ast declarations and only use it as normal function internally?

Reply:
this is so the code how it was previously written is still recognizable the same way now: `parseOr(tokens, parseFoo(parseBar(...)))` how it was used in the manual switch-case parsing code before now simply became `parseOr(parseFoo(parseBar(...)))` inside the declaration. I don't think we will ever have anything else than or on top level, but to match the previous code I kept it like that. However if we want to diverge from the original code more we can remove the unused / repeating stuff like this.

Reply:
The old code's structure was to mimic rust's, but this PR abandons that anyway.
So I'd prefer it without the meaningless `parseOr` everywhere.